### PR TITLE
NODE-2291 Add request span metrics

### PR DESCRIPTION
**Pull request by ivan-mashonskiy**

### CODE REVIEW

File: node/build.sbt [hunk 1]
```diff
@@ -1,9 +1,26 @@
...
+  GitVersioning,
+  VersionObject,
+  JavaAgent
+)
 
 libraryDependencies ++= Dependencies.node.value
 
+javaAgents ++= {
+  if (instrumentation.value) {
+    Seq("io.kamon" % "kanela-agent" % "1.0.14")
```
Comment (vsuharnikov) [inline node/build.sbt]: Probably, `Dependencies` is a better place for this

File: node/src/main/scala/com/wavesplatform/Application.scala [hunk 2]
```diff
@@ -327,19 +336,45 @@ class Application(val actorSystem: ActorSystem, val settings: WavesSettings, con
           "rest-time-limited",
           reporter = log.trace("Uncaught exception in time limited pool", _)
         )
+      val heavyRequestProcessorPoolThreads =
+        settings.restAPISettings.heavyRequestProcessorPoolThreads.getOrElse((Runtime.getRuntime.availableProcessors() * 2).min(4))
```
Comment (vsuharnikov) [inline node/src/main/scala/com/wavesplatform/Application.scala:340]: Could you add a comment, why do we need ~at least~ maximum 4 threads?

File: node/src/main/scala/com/wavesplatform/Application.scala [hunk 3]
```diff
@@ -327,19 +336,45 @@ class Application(val actorSystem: ActorSystem, val settings: WavesSettings, con
...
+      val heavyRequestExecutor = new ThreadPoolExecutor(
+        heavyRequestProcessorPoolThreads,
+        heavyRequestProcessorPoolThreads,
+        60,
+        TimeUnit.SECONDS,
```
Comment (vsuharnikov) [inline node/src/main/scala/com/wavesplatform/Application.scala]: I know, these values doesn't affect a number of threads in the pool in this case, but probably it'd be better to specify 0 millis as `Executors.newFixedThreadPool` does:
```java
public static ExecutorService newFixedThreadPool(int nThreads, ThreadFactory threadFactory) {
    return new ThreadPoolExecutor(nThreads, nThreads,
                                  0L, TimeUnit.MILLISECONDS,
                                  new LinkedBlockingQueue<Runnable>(),
                                  threadFactory);
}
```

File: node/src/main/scala/com/wavesplatform/api/http/AddressApiRoute.scala [hunk 4]
```diff
@@ -287,19 +286,22 @@ case class AddressApiRoute(
...
+    )(_.runAsyncLogErr(_))
+  }
 
-  private def accountData(addr: Address, regex: String)(implicit sc: Scheduler) =
-    commonAccountsApi
-      .dataStream(addr, Some(regex))
-      .toListL
-      .runAsyncLogErr
-      .map(data => Source.fromIterator(() => data.sortBy(_.key).iterator.map(Json.toJson[DataEntry[?]])))
+  private def accountData(addr: Address, regex: String)(implicit m: ToResponseMarshaller[Source[JsValue, NotUsed]]) =
```
Comment (vsuharnikov) [inline node/src/main/scala/com/wavesplatform/api/http/AddressApiRoute.scala]: The only difference of these two `accountData` methods is `regex: String`. How about:
```scala
private def accountData(addr: Address, regex: Option[String] = None)(implicit m: ToResponseMarshaller[Source[JsValue, NotUsed]])
```
?

File: node/src/main/scala/com/wavesplatform/api/http/BlocksApiRoute.scala [hunk 5]
```diff
@@ -17,8 +17,9 @@ import com.wavesplatform.transaction.TxValidationError.GenericError
 import com.wavesplatform.utils.Time
 import play.api.libs.json.*
 
-case class BlocksApiRoute(settings: RestAPISettings, commonApi: CommonBlocksApi, time: Time) extends ApiRoute {
+case class BlocksApiRoute(settings: RestAPISettings, commonApi: CommonBlocksApi, time: Time, routeTimeout: RouteTimeout) extends ApiRoute {
   import BlocksApiRoute.*
+  private[this] val MaxBlocksPerRequest = 100 // todo: make this configurable and fix integration tests
```
Comment (vsuharnikov) [inline node/src/main/scala/com/wavesplatform/api/http/BlocksApiRoute.scala]: I remind you about this todo :)

File: node/src/main/scala/com/wavesplatform/Application.scala (hunk 2 above)
Comment (xrtm000) [reply node/src/main/scala/com/wavesplatform/Application.scala:340]: And why doubled available processors?

File: node/src/main/scala/com/wavesplatform/Application.scala [hunk 6]
```diff
@@ -390,19 +428,36 @@ class Application(val actorSystem: ActorSystem, val settings: WavesSettings, con
...
+        ),
         RewardApiRoute(blockchainUpdater)
       )
 
       val httpService = CompositeHttpService(apiRoutes, settings.restAPISettings)
       val httpFuture =
         Http().newServerAt(settings.restAPISettings.bindAddress, settings.restAPISettings.port).bindFlow(httpService.loggingCompositeRoute)
       serverBinding = Await.result(httpFuture, 20.seconds)
-      serverBinding.whenTerminated.foreach(_ => httpService.scheduler.shutdown())
+      serverBinding.whenTerminated.foreach(_ => heavyRequestScheduler.shutdown())
```
Comment (xrtm000) [inline node/src/main/scala/com/wavesplatform/Application.scala:460]: Why `httpService.scheduler` should not be shutdowned?

File: node/src/main/scala/com/wavesplatform/api/common/CommonAccountsApi.scala [hunk 7]
```diff
@@ -57,9 +57,8 @@ object CommonAccountsApi {
 
   def apply(diff: () => Diff, db: DB, blockchain: Blockchain): CommonAccountsApi = new CommonAccountsApi {
 
-    override def balance(address: Address, confirmations: Int = 0): Long = {
-      blockchain.balance(address, blockchain.height, confirmations)
-    }
+    override def balance(address: Address, confirmations: Int = 0): Long =
+      blockchain.balance(address)
```
Comment (xrtm000) [inline node/src/main/scala/com/wavesplatform/api/common/CommonAccountsApi.scala]: Why confirmations should be unused?

**Review by DrBlast — APPROVED** [review]

**Review by xrtm000 — APPROVED** [review]

**Review by vsuharnikov — APPROVED** [review]

**Review by phearnot — APPROVED** [review]

**Pull request merged**